### DISCUSSION TITLE
Revoir les critères

### DISCUSSION THREAD

**Discussion by polatouche-spip** [cnumr/EcoIndex#230] [Ideas]
J'aimais bien la version [d'Ecometer ](http://www.ecometer.org) initiale, à quelques détails près
facilement corrigeables, et suis très surpris de cette nouvelle version
qui me semble aller dans une mauvaise direction.

D'abord, l'affirmation qu'un grand nombre requêtes provoque un grand nombre
de serveurs est factuellement fausse: ce n'est vrai que si les domaines
sont différents, ce qui arrive peu souvent, et c'est justement pour cela
qu'ont été développé HTTP2 et QUICK. Ce qu'il faut sanctionner ici c'est:
- la multiplicité des domaines dans les requêtes
- à l'inverse des requêtes vers le même serveur pour des données de même type (JS, CSS etc) car elle pourraient être fusionnées.
Ensuite, une bonne analyse multi-critères exige que les critères soient 
décorrélés, ce qui n'est pas le cas ici: le poids de la page englobe 
le nombre d'élements du DOM. Et de l'affirmation que un grand nombre d'éléments
va induire un gros calcul est grossière: une page HTML réduite à une table HTML
même grande est plus facile à mettre en page qu'un nombre réduit d'éléments
mais très imbriqués, car le calcul de la cascade de styles va être compliqué.
Le bon critère ici est donc la profondeur d'imbrication, pas le nombre.

Ensuite, il est surprenant que la présence de JavaScript et le volume 
qu'il représene ne soit pas pris en compte, alors que ce sont les bibliothèques
JavaScript qui sont largement responsable du surcoût des pages Web depuis 
leur apparition: sans JavaScript, le serveur HTTP calcule le code HTML une
fois pour toutes pour tous ses clients, avec JavaScript chacun d'eux doit
synthétiser ce code, alors que bien souvent le résultat sera le même.
Il serait plus utile de mesurer le temps entre la fin de la requête
initiale et l'affichage complet de la page, ou au moins le volume de Javascript
chargé.

Enfin, je suis surpris que les critères de la précédente version
(présence de minification, analyse des en-têtes HTTP etc) aient disparu:
c'était des indications importantes qui suggéraient des pistes possibles.
Les critères de la nouvelle version sont trop généraux pour inciter à 
améliorer rapidement.

**Comment by tbroyer** [comment]
EcoIndex n'est pas Ecometer, ce sont des projets différents (et EcoIndex est le plus ancien des deux)

**Reply by polatouche-spip** [reply]
Depuis longtemps le site Ecometer part en vrille quand on essaye de l'utiliser, j'ai donc pensé qu'il était abandonné pour celui-ci. Que cela ne soit pas le cas, soit, mais pourquoi démarrer un nouveau projet plutôt que d'améliorer un projet déjà existant et déjà plus abouti ?
